src/user_input.rs

```diff
@@ -47,6 +56,38 @@ impl UserInput {
         }
     }
 
+    /// Creates a [`UserInput::Chord`] from an iterator of [`InputKind`]s
+    ///
+    /// If `inputs` has a length of 1, a [`UserInput::Single`] variant will be returned instead.
+    ///
+    /// If `inputs` contains more than `CHORD_MAX_SIZE` elements, a [`ChordCreationError`] is returned.
+    /// If you want to silently ignore the rest of the elements, use the [`UserInput::chord`] function instead.
+    pub fn try_make_chord(
```

Comment:
Maybe we can use something super generic here? Like `new` or `try_create` or something?

This feels like it should be the default method for a lot of things.